src/ocean/net/server/connection/IFiberConnectionHandler.d

```diff
@@ -221,6 +221,25 @@ abstract class IFiberConnectionHandlerBase : IConnectionHandler
 
     /***************************************************************************
 
+        Called by `finalize` to unregister the connection socket from epoll
+        before closing it. This is done because closing a socket does not always
+        mean that it is unregistered from epoll -- in situations where the
+        process has forked, the fork's reference to the fd will prevent it from
+        being unregistered until the fork exits. Therefore, to be certain that
+        the socket will not fire again in epoll, we need to explicitly
+        unregister it.
+
+    ***************************************************************************/
+
+    override protected void unregisterSocket ( )
+    {
+        if ( auto registered_client = this.fiber.registered_client )
+            if ( this.socket.fd == registered_client.fileHandle() )
```

Comment:
What would be the reason when this is not true?

Reply:
It would mean that the connection fiber is suspended waiting for another event, not I/O on the socket. (Thus the socket is not registered with epoll.)